### PR TITLE
Fix nil dereference panic on request timeout

### PR DESCRIPTION
**Pull request by nilebox**
Fix for the bug https://github.com/kubernetes-incubator/service-catalog/issues/905

### CODE REVIEW

File: pkg/brokerapi/openservicebroker/open_service_broker_client.go [hunk 1]
```diff
@@ -148,7 +148,11 @@ func (c *openServiceBrokerClient) CreateServiceInstance(ID string, req *brokerap
 	)
 	if err != nil {
 		glog.Errorf("Error sending create service instance request to broker %q at %v: response: %v error: %#v", c.name, serviceInstanceURL, resp, err)
-		return nil, resp.StatusCode, errRequest{message: err.Error()}
```
Comment (nilebox) [inline pkg/brokerapi/openservicebroker/open_service_broker_client.go]: I checked the code invoking CreateServiceInstanceResponse() method, and it's actually never checks status if the error occured. So the alternative fix is to just
```
return nil, nil, errRequest{message: err.Error()}
```

File: pkg/brokerapi/openservicebroker/open_service_broker_client.go [hunk 2]
```diff
@@ -148,7 +148,11 @@ func (c *openServiceBrokerClient) CreateServiceInstance(ID string, req *brokerap
...
+		errReq := errRequest{message: err.Error()}
+		if resp == nil {
+			return nil, nil, errReq
```
Comment (duglin) [inline pkg/brokerapi/openservicebroker/open_service_broker_client.go]: s/nil/0/ for the 2nd one since you can't return nil for an int.

Comment (nilebox) [inline pkg/brokerapi/openservicebroker/open_service_broker_client.go]: whoops sorry

**Comment by arschles** [comment]
This looks good @nilebox, thanks for it. I'm merging now

**Pull request merged**